libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -52,6 +54,109 @@ static int lua_millis(lua_State *L) {
     return 1;
 }
 
+static int lua_mavlink_receive(lua_State *L) {
+    check_arguments(L, 0, "receive");
+
+    struct AP_Scripting::mavlink_msg msg;
+    ObjectBuffer<struct AP_Scripting::mavlink_msg> *input = AP::scripting()->mavlink_data.input;
+
+    if (input == nullptr) {
+        luaL_error(L, "Never subscirbed to a message");
+    }
+
+    if (input->pop(msg)) {
+        luaL_Buffer b;
+        luaL_buffinit(L, &b);
+        luaL_addlstring(&b, (char *)&msg.msg, sizeof(msg.msg));
+        luaL_pushresult(&b);
+        lua_pushinteger(L, msg.chan);
+        return 2;
+    } else {
+        // no MAVLink to handle, just return no results
+        return 0;
+    }
+}
+
+static int lua_mavlink_register_msgid(lua_State *L) {
+    check_arguments(L, 1, "register_msgid");
+    luaL_checkstack(L, 1, "Out of stack");
+    
+    const int msgid = luaL_checkinteger(L, -1);
+    luaL_argcheck(L, ((msgid >= 0) && (msgid < (1 << 24))), 1, "msgid out of range");
+
+    struct AP_Scripting::mavlink &data = AP::scripting()->mavlink_data;
+
+    if (data.input == nullptr) {
+        { // WITH_SEMAPHORE cannot be used in the same scope as a luaL_error (or any thing else that jumps away)
+            WITH_SEMAPHORE(data.sem);
+            data.input = new ObjectBuffer<struct AP_Scripting::mavlink_msg>(5);
+        }
+        if (data.input == nullptr) {
+            return luaL_error(L, "Unable to allocate MAVLink buffer");
+        }
+    }
+
+    // check that we aren't currently watching this ID
+    for (int id : data.accept) {
+        if (id == msgid) {
+            lua_pushboolean(L, false);
+            return 1;
+        }
+    }
+
+    int i = 0;
+    for (i = 0; i < (int)ARRAY_SIZE(data.accept); i++) {
+        if (data.accept[i] == -1) {
+            break;
+        }
+    }
+
+    if (i >= (int)ARRAY_SIZE(data.accept)) {
+        return luaL_error(L, "Out of MAVLink ID's to monitor");
+    }
+
+    {
+        WITH_SEMAPHORE(data.sem);
+        data.accept[i] = msgid;
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+static int lua_mavlink_send(lua_State *L) {
+    check_arguments(L, 3, "send");
+    
+    const mavlink_channel_t chan = (mavlink_channel_t)luaL_checkinteger(L, 1);
+    luaL_argcheck(L, (chan < MAVLINK_COMM_NUM_BUFFERS), 1, "channel out of range");
+
+    const int msgid = luaL_checkinteger(L, 2);
+    luaL_argcheck(L, ((msgid >= 0) && (msgid < (1 << 24))), 2, "msgid out of range");
+
+    const char *packet = luaL_checkstring(L, 3);
+
+    struct AP_Scripting::mavlink &mavlink_data = AP::scripting()->mavlink_data;
+
+    if (mavlink_data.output == nullptr) {
+        mavlink_data.output = new ObjectBuffer<struct AP_Scripting::mavlink_output>(2);
```

Comment:
Again, a rather odd place to put a constant.  At least have a `const` variable to pull it out as something reasonably important.